lib/cocoapods/installer/analyzer.rb

```diff
@@ -170,6 +170,9 @@ def sources
           end
           unless plugin_sources.empty?
             result.insert(0, *plugin_sources)
+            plugin_sources.each do |source|
+              config.sources_manager.add_source(source)
```

Comment:
should this be added to the global sources manager? should we make our own sources manager in the analyzer?

Reply:
updated this to use a separate sources manager rather than the global one